test/unit/specs/data/json-update-builder.js

```diff
@@ -0,0 +1,171 @@
+import { describe, it } from 'mocha'
+
+import Backendless from '../../helpers/sandbox'
+import { expect } from 'chai'
+
+describe('<Data> JSON Update Builder', function() {
+  let jsonUpdateBuilder
+
+  beforeEach(() => {
+    jsonUpdateBuilder = Backendless.Data.JSONUpdateBuilder
+  })
+
+  it('check default values', async () => {
+    const jsonUpdateBuilderInstance = new jsonUpdateBuilder()
+
+    expect(jsonUpdateBuilderInstance.args).to.be.eql({})
+    expect(jsonUpdateBuilderInstance.operationName).to.be.equal(undefined)
+  })
+
+  it('set', async () => {
+    const jsonUpdateSet = jsonUpdateBuilder.SET()
+
+    expect(jsonUpdateSet.args).to.be.eql({})
+    expect(jsonUpdateSet.operationName).to.be.equal('JSON_SET')
+  })
+
+  it('set with args', async () => {
+    const jsonUpdateSet = jsonUpdateBuilder.SET()
+      .addArgument('$.letter', 'b')
+      .addArgument('$.number', 36)
+      .addArgument('$.state', true)
+      .addArgument('$.colours[0]', null)
+      .addArgument('$.innerObject', { a: 'b' })
+      .addArgument('$.innerArray', [4, 3, 2])
+
+    expect(jsonUpdateSet.args).to.eql({
+      '$.letter'     : 'b',
+      '$.number'     : 36,
+      '$.state'      : true,
+      '$.colours[0]' : null,
+      '$.innerObject': { a: 'b' },
+      '$.innerArray' : [4, 3, 2]
+    })
+    expect(jsonUpdateSet.operationName).to.be.equal('JSON_SET')
+  })
+
+  it('set with args and run create()', async () => {
+    const jsonUpdateSet = jsonUpdateBuilder.SET()
```

Comment:
also need to check how `toJSON` works inside the request

```
// register a mock request

 const jsonUpdateReplace = jsonUpdateBuilder.REPLACE()
      .addArgument('$.decimals[2]', 20)
      .addArgument('$.decimals[3]', 25)
      .addArgument('$.state', 'on')
      .addArgument('$.number', 11)

    const newValues = { myJson: jsonUpdateReplace, objectId: 'test-id' }

    const updateResult = await TestTable.save(newValues)

   // check request body of the mock request
```

Reply:
+